### ISSUE TITLE
Reset Function keyframes

### ISSUE THREAD

**Issue by jeremiahmmv2** [jeremyckahn/rekapi#36]
I had a problem when integrating Rekapi into a slider that can be looped. When I let the animations finish for each slide, everything worked as it should when the slider looped back around. However, the slider lets the user skip before the animation has completed. When the user skips a slide after a function keyframe fires, but before the last function keyframe fires, the first function keyframe would not fire when the slider looped back around. In fact, the only function keyframe that fires is the last one.

It took me a few hours digging through the Rekapi code to realize function frames are reset by setting hasFired to FALSE.

It would be nice if there were an API method to explicitly reset function keyframes. This is a rough example, but it works in my implementation:

```javascript
  Rekapi.prototype.resetFunctionKeyframes = function () {
    var rekapi = this;

    var lookupObject = { name: 'function' };
    _.each(rekapi._actors, function (actor) {
        var fnKeyframes = _.where(actor._keyframeProperties, lookupObject);
        
        _.each(fnKeyframes, function (fnKeyframe) {
          fnKeyframe.hasFired = false;
        });
    });
  }
```

**Comment by jeremyckahn** [comment]
Hi @jeremiahmmv2, thanks for reporting this.  Yes, function keyframes are reset internally, as you discovered.  This is necessary to account for the "stateless" nature of a given animation frame within Rekapi.  Rekapi needs to know whether or not a function keyframe has executed for a given loop iteration.  Admittedly this isn't ideal, but it works well enough for common cases and is necessary to prevent repeated calls to a keyframe's function.

I think your approach of explicitly resetting the function keyframes would be a useful addition to Rekapi.  Ideally this shouldn't even be necessary — Rekapi should just be a bit smarter about resetting function keyframes automatically.  I'd like to think on this for a few days, but I will probably add your solution to Rekapi if I can't come up with something a little less manual.  I will update this ticket when I have something.  Thanks again!

**Comment by jeremyckahn** [comment]
@jeremiahmmv2, I have a potential fix in my `bugfix/36-reset-later-keyframes` branch.  Can you please pull it down and see if it fixes your issue without having to explicitly call your `Rekapi.prototype.resetFunctionKeyframes` method?  You will need to run `grunt build` locally to get the changes into `dist` before testing.  Thanks!

**Comment by jeremyckahn** [comment]
I forgot something, and just added it.  Please give my `bugfix/36-reset-later-keyframes` branch a try!

**Comment by jeremiahmmv2** [comment]
I wasn't familiar with grunt, but now I am. :)

The `bugfix/36-reset-later-keyframes` branch works just as expected. Thanks!

**Comment by jeremyckahn** [comment]
This is fixed in [1.4.7](https://github.com/jeremyckahn/rekapi/releases/tag/1.4.7).  Thanks again for reporting!
